src/Types/Type.php

```diff
@@ -141,10 +138,16 @@ public static function lookupName(self $type): string
      * @param class-string<Type> $className The class name of the custom type.
      *
      * @throws Exception
+     *
+     * @deprecated Use Type::getTypeRegistry()->register() instead.
      */
     public static function addType(string $name, string $className): void
```

Comment:
We could keep this method if the arg type were widened.
```suggestion
    public static function addType(string $name, string|Type $type): void
```

Either we keep this API, or we deprecate all the static methods of this class (`getType`, `hasType`, `lookupName`, `overrideType`, `getTypesMap`), which would have a large impact on the Doctrine related packages.

If you don't want to change the signature of this method, you can add `setType(string $name, Type $type, bool $override = false)`. 

Note that the `DoctrineBundle` would use the optional "override" in [`ConnectionFactory`](https://github.com/doctrine/DoctrineBundle/blob/a5f561399328802af855e187f1786a918c35e837/src/ConnectionFactory.php#L208-L212).

Reply:
I like it, I pushed the changes. Much simpler now!